LayoutKit/Layouts/ButtonLayout.swift

```diff
@@ -0,0 +1,289 @@
+        case .contactAdd, .infoLight, .infoDark, .detailDisclosure:
+            // Setting a custom font has no effect in this case.
+            return defaultFontForSystemButton
+        }
+    }
+
+    private var defaultFontForCustomButton: UIFont {
+        #if os(tvOS)
+            return UIFont.systemFont(ofSize: 38, weight: UIFontWeightMedium)
+        #else
+            return UIFont.systemFont(ofSize: 18)
+        #endif
+    }
+
+    private var defaultFontForSystemButton: UIFont {
+        #if os(tvOS)
+            return UIFont.systemFont(ofSize: 38, weight: UIFontWeightMedium)
+        #else
+            return UIFont.systemFont(ofSize: 15)
+        #endif
+    }
+
+    private func sizeOf(image: ButtonLayoutImage) -> CGSize {
+        switch image {
+        case .size(let size):
+            return size
+        case .image(let image):
+            return image?.size ?? .zero
+        case .defaultImage:
+            switch type {
+            case .custom, .system:
+                return .zero
+            case .contactAdd, .infoLight, .infoDark, .detailDisclosure:
+                #if os(tvOS)
+                    return CGSize(width: 37, height: 46)
+                #else
+                    return CGSize(width: 22, height: 22)
+                #endif
+            }
+        }
+    }
+
+    private var minWidth: CGFloat {
+        switch type {
+        case .custom, .system:
+            return hasCustomStyle ? 0 : 30
+        case .detailDisclosure, .infoLight, .infoDark, .contactAdd:
+            return 0
+        }
+    }
+
+    private var hasCustomStyle: Bool {
+        return hasCustomImage || contentEdgeInsets != .zero
+    }
+
+    private var hasCustomImage: Bool {
+        switch image {
+        case .size, .image:
+            return true
+        case .defaultImage:
+            return false
+        }
+    }
+
+    private var verticalPadding: CGFloat {
+        switch type {
+        case .custom, .system:
+            return hasCustomStyle ? 0 : 12
+        case .detailDisclosure, .infoLight, .infoDark, .contactAdd:
+            #if os(tvOS)
+                return isTitleEmpty && !hasCustomImage ? -9 : 0
+            #else
+                return 0
+            #endif
+        }
+    }
+
+    private var isTitleEmpty: Bool {
+        switch title {
+        case .unattributed(let text):
+            return text == ""
+        case .attributed(let text):
+            return text.string == ""
+        }
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    open override func makeView() -> View {
+        return Button(type: type.buttonType)
+    }
+
+    open override func configure(view: Button) {
+        config?(view)
+        view.contentEdgeInsets = contentEdgeInsets
```

Comment:
No because `titleEdgeInsets` and `imageEdgeInsets` don't affect the actual layout. The doc for both says.

> This property is used only for positioning the title during layout. The button does not use this property to determine intrinsicContentSize and sizeThatFits(_:).

You can customize these in the config block but it doesn't change the size or position of the UIButton.